nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1817,171 @@ def __init__(
+
+    """
+
+    no_parallelization = True
+    recast = True
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        *,
+        init_budget: tp.Optional[int] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+
+        from bayes_optim.extension import PCABO as PcaBO
+        from bayes_optim.extension import RealSpace
+
+        lb, ub = 1e-7 - np.pi / 2, np.pi / 2 - 1e-7
+        space = RealSpace([lb, ub]) * self.dimension
+        self.buffer: List[float] = []
+        self.newX: List[float] = []
+        self.loss: List[float] = []
+
+        self._pcabo = PcaBO(
+            search_space=space,
+            obj_fun=None,  # Assuming that this is not used :-)
+            DoE_size=init_budget if init_budget is not None else 5,
+            max_FEs=budget,
+            verbose=True,
+            n_point=1,  # We start with a sequential procedure, maybe we'll extend in a second moment
+            n_components=0.95,
+            acquisition_optimization={"optimizer": "BFGS"},
+        )
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        if not self.buffer:
+            self.newX = []
+            self.loss = []
+            candidate = self._pcabo.ask()
+            if not isinstance(candidate, list):
+                candidate = candidate.tolist()
+            self.buffer = candidate
+        x_probe = self.buffer.pop()
+        print("X_PROBE = ", x_probe)
+        print("BUFFER = ", self.buffer)
+        # data = self._pcabo._transform.backward(np.array(x_probe, copy=False))
+        data = np.tan(np.array(x_probe, copy=False))
+        candidate = self.parametrization.spawn_child().set_standardized_data(data)
+        candidate._meta["x_probe"] = x_probe
+        self.newX.append(candidate._meta["x_probe"])
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        # print("BUFFER inside tell = ", self.buffer)
+        print("newX =", self.newX)
+        self.loss.append(loss)
+        print("new_loss =", self.loss)
+        if not self.buffer:
+            if "x_probe" in candidate._meta:
+                # y = candidate._meta["x_probe"]
+                print("Len(newX) = ", len(self.newX))
+                self._pcabo.tell(self.newX, self.loss)
+            else:
+                data = candidate.get_standardized_data(reference=self.parametrization)
+                # Tell not asked:
+                self._pcabo.tell(np.arctan(data), loss)
+
+    def _internal_tell_not_asked(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        raise errors.TellNotAskedNotSupportedError
+
+
+@registry.register
+class BayesOptimBO(base.Optimizer):
+    """BO algorithm from BayesOptim package"""
+
+    no_parallelization = True
+    recast = True
+
+    def __init__(
+        self,
```

Comment:
what is different in this version? This looks mostly like a copy paste, which must be avoided (it's always a pain to maintain on the long run).

Reply:
The algorithm comes from bayes-optim package but is BO rather than PCA-BO. Yes, it is a copy of the other one, which calls a different optimizer and tunes the hyperparameters of the GPR model - which is automatically done in the case of PCA-BO.

Reply:
so the only difference is `self._bo2` instead of `self._pcabo`?
what about adding a parameter `pca: bool` in `ParametrizedPCABO`, and then instantiate `self._alg` with either one of the option?

Reply:
Not really.. there is also the tuning of the hyperparameters of the GPR model and some other minor things. But I can merge the two classes.

Reply:
Done